opensrp-app/src/test/java/org/smartregister/sync/helper/LocationServiceHelperTest.java

```diff
@@ -274,4 +290,70 @@ public void testFetchAllLocations() {
         String requestString = stringArgumentCaptor.getAllValues().get(1);
         assertEquals("{\"is_jurisdiction\":true,\"serverVersion\":0}", requestString);
     }
+
+    @Test
+    public void testGetFormattedBaseUrlStripsEndingSlashDelimiter() {
+        String url = "https://base.url/";
+
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", coreLibrary);
+        Mockito.doReturn(context).when(coreLibrary).context();
+        Mockito.doReturn(configuration).when(context).configuration();
+        Mockito.doReturn(url).when(configuration).dristhiBaseURL();
+
+        LocationServiceHelper spyLocationServiceHelper = Mockito.spy(locationServiceHelper);
+        String baseUrl = spyLocationServiceHelper.getFormattedBaseUrl();
+        assertNotNull(baseUrl);
+        assertEquals(url.substring(0, url.length() - 1), baseUrl);
+    }
+
+    @Test
+    public void testGetLocationHierarchyReturnsNullWhenRequestFails() {
+        String parentLocation = "1c7ba751-35e8-4b46-9e53-3cb8fd193697";
+
+        Mockito.doReturn(new Response<>(ResponseStatus.failure, ""))
+                .when(httpAgent).fetch(stringArgumentCaptor.capture());
+
+        LocationTree tree = locationServiceHelper.getLocationHierarchy(parentLocation);
+
+        String expectedUrl = MessageFormat.format("{0}{1}{2}", locationServiceHelper.getFormattedBaseUrl(), LocationServiceHelper.LOCATION_HIERARCHY_URL, parentLocation);
+        assertEquals(expectedUrl, stringArgumentCaptor.getAllValues().get(0));
+        assertNull(tree);
+    }
+
+    @Test
+    public void testGetLocationHierarchyReturnsNullForInvalidParentLocationId() {
```

Comment:
Doesn't seem like the method returns null for invalid parent location id? https://github.com/OpenSRP/opensrp-client-core/pull/663/files#diff-7d2c1e8994b4bf118233996655641ad90f6f71d2f8145fcea0e1bf67136a37e7R334

Reply:
Updated test method name to reflect the actual test